packages/es-components/src/components/controls/buttons/IconButton.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Icon from '../../base/icons/Icon';
+
+const ContainerButton = styled.button`
+  align-items: center;
+  background-color: transparent;
+  border: none;
+  cursor: ${props => (props.disabled ? '' : 'pointer')};
+  display: flex;
+  flex-direction: column;
+  font-family: inherit;
+
+  i {
+    background-color: ${props =>
+      props.isHighlighted ? props.theme.brandColors.primary3 : 'white'};
+    color: ${props =>
+      props.isHighlighted ? 'white' : props.theme.brandColors.primary3};
+  }
+  &:active:enabled i {
+    background-color: ${props =>
+      !props.isHighlighted ? props.theme.brandColors.primary3 : 'white'};
+    color: ${props =>
+      !props.isHighlighted ? 'white' : props.theme.brandColors.primary3};
+  }
+
+  span {
+    font-weight: ${props => (props.isHighlighted ? 'bold' : 'normal')};
+  }
+  &:active:enabled span {
+    font-weight: ${props => (!props.isHighlighted ? 'bold' : 'normal')};
+  }
+`;
+const IconWithBorder = styled(Icon)`
+  border: 2px solid;
+  border-color: ${props => props.theme.brandColors.primary3};
+  border-radius: 50%;
+  padding: 5px;
+`;
+const ChildrenSpan = styled.span`
+  color: ${props => props.theme.colors.gray9};
+  font-size: ${props => props.fontSize}px;
+  max-width: ${props => props.maxWidth};
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
+
+function IconButton({
+  iconName,
+  iconSize,
+  childrenFontSize,
+  isHighlighted,
+  disabled,
+  maxWidth,
+  onClick,
+  children
+}) {
+  return (
+    <ContainerButton
+      onClick={onClick}
+      isHighlighted={isHighlighted}
+      disabled={disabled}
```

Comment:
We should make sure to include any extra properties (`{...otherProps}`) to make sure we can add additional styles to this from outside. That'll allow `className` and `style` props to be applied downstream.

Reply:
That's a good idea! I overlooked that.

Reply:
Done!